packages/components/Menu/src/Menu/Menu.tsx

```diff
@@ -21,20 +23,41 @@ export const Menu = stagedComponent((props: MenuProps) => {
         console.warn('Menu must contain two children');
       }
     }
-
-    const menuTrigger = childrenArray[0];
-    const menuPopover = childrenArray[1];
-
-    return (
-      <MenuProvider value={contextValue}>
-        {menuTrigger}
-        {/* GH#2661: Make sure that shouldFocusOnContainer is defined before initializing
-            the popover so that focus is put in the correct place */}
-        {state.open && state.shouldFocusOnContainer !== undefined && menuPopover}
-      </MenuProvider>
-    );
+    return renderFinalMenu(childrenArray, contextValue, state);
   };
 });
+
+const renderFinalMenu = (childrenArray: React.ReactElement[], contextValue: MenuContextValue, state: MenuState) => {
+  const menuTrigger = childrenArray[0];
+  const menuPopover = childrenArray[1];
+  return (
+    <MenuProvider value={contextValue}>
+      {Platform.OS === 'android' ? (
+        <React.Fragment>
+          <View
+            onLayout={(event) => {
+              const { width } = event.nativeEvent.layout;
+              state.setAnchorWidth(width);
+            }}
```

Comment:
This can split into a separate per platform component - there is no code sharing in this code path,

Reply:
It was split per platform in the previous version of the PR but as the outer render code is same for all the platform , Ruriko's comment was to share them, hence added it here.

Reply:
Yeah, we should share the outer portion across platforms, but the renderFInalMenu function should be separate per platform.

So it'd be something like:

Menu.tsx:
stagedComponent

renderMenu.tsx
renderFinalMenu

renderMenu.android.tsx:
renderFinalMenu

This allows the outer portion to be shared while the inner portion would be split between platforms.